nexus/types/src/external_api/params.rs

```diff
@@ -276,7 +276,7 @@ pub struct SiloCreate {
 pub struct UserCreate {
```

Comment:
What do we want the JSON to look like? Today it's something like this:
```json
{
  "external_id": "xyz",
  "password": {
    "mode": "password",
    "value": "l1admin"
  }
}

{
  "external_id": "xyz",
  "password": {
    "mode": "invalid_password"
  }
}
```

That seems good. Is this any better?

```json
{
  "external_id": "xyz",
  "password": "l1admin"
}

{
  "external_id": "xyz",
}
```

The absence of `"password"` or a value of `null` would indicate that the password should be unset? It's the best alternative I can think of, and I hate it.

Or maybe:
```json
{
  "external_id": "xyz",
  "mode": "password",
  "value": "l1admin"
}

{
  "external_id": "xyz",
  "mode": "invalid_password"
}
```

... but that means we'd have a Rust type like this:

```rust
#[serde(tag = "mode")]
pub enum UserCreate {
    Password {
        pub external_id: UserId,
        pub value: String,
    }
    InvalidPassword {
        pub external_id: UserId,
    }
}
```

Enumeration options to say: I like what you've chosen and can't think of something better